monix-reactive/shared/src/main/scala/monix/reactive/Observable.scala

```diff
@@ -1751,6 +1751,28 @@ abstract class Observable[+A] extends Serializable { self =>
   final def map[B](f: A => B): Observable[B] =
     self.liftByOperator(new MapOperator(f))
 
+  /** Applies a function that you supply to each item emitted by the
+    * source observable, where that function returns a sequence of elements, and
+    * then concatenating those resulting sequences and emitting the
+    * results of this concatenation.
+    *
+    * ==Example==
+    * {{{
+    *   Observable(1, 2, 3).mapConcat( x => Seq(x, x * 10, x * 100))
+    * }}}
+    *
+    * == Visual Example ==
+    *
+    * <pre>
+    * stream: 1 -- -- 2 -- -- 3 -- --
+    * result: 1, 10, 100, 2, 20, 200, 3, 30, 300
+    * * </pre>
+    *
+    * @param f is a generator for the sequences being concatenated
+    */
+  final def mapConcat[B](f: A => Seq[B]): Observable[B] =
```

Comment:
I'm wondering about two things:
1. Should we use `Iterable` instead of `Seq`? `Seq extends Iterable` but it also introduce some dangerous operations like `size` or `def apply(i: Int): A` and is usually discouraged. On the other hand it is used in many other existing operators...

2. We also have `concatMap` which is an alias for `flatMap`. `concatMap` comes from RX but they don't seem to have anything named `mapConcat` which is consistent with Akka. I guess it's fine then, I'm only slightly worried about potential confusion.

Reply:
1. I think we should `Iterable` as well, I will go for `scala.collection.immutable.Iterable` that's what makes more sense and after checking that is also exactly what akka-streams is using.

2. For `concatMap` vs `mapConcat` i have to admit it might be a bit confusing but hopefully the signature and doc should help to allievate the confusion

Reply:
Pushed the changes for using scala.collection.immutable, only remaining is the property based test